Sources/SwiftDocC/Semantics/Metadata/PageColor.swift

```diff
@@ -0,0 +1,69 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2023 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import Markdown
+
+/// A directive that allows you to set the color that should be used to
+/// represent a documentation page.
+///
+/// Use the `PageColor` directive to provide a hint to the Swift-DocC renderer as to what color
+/// should represent this page. The renderer will use this color, depending on the context,
+/// as a foundation for the other accent colors used in the page. For example, Swift-DocC-Render
+/// will use this color as the primary background color of a page's introduction section
+/// and adjust the colors of the icon and text in the page's introduction accordingly.
+///
+/// This directive is only valid within a ``Metadata`` directive:
+///
+/// ```markdown
+/// @Metadata {
+///     @PageColor(orange)
+/// }
+/// ```
+public final class PageColor: Semantic, AutomaticDirectiveConvertible {
+    public let originalMarkup: BlockDirective
+    
+    /// A context-dependent, standard color.
+    @DirectiveArgumentWrapped(name: .unnamed)
+    public var color: Color
+    
+    /// A context-dependent, standard color.
+    public enum Color: String, CaseIterable, DirectiveArgumentValueConvertible {
```

Comment:
Might be worth defining these somewhere more central and extending them to conform to `DirectiveArgumentValueConvertible` here. the advantage would be that we wouldn't have to stringify these values until render node encoding time.

Reply:
I'd rather leave the RenderNode implementation more string-based for now until we have a better idea of how we'll implement longer-term support for custom page types. Our current needs are simple enough that I think we can prioritize flexibility here and reduce the risk of breaking API later on.